x-pack/platform/plugins/shared/lens/public/datasources/form_based/utils.tsx

```diff
@@ -339,8 +348,11 @@ export function getUnsupportedOperationsWarningMessage(
     [FieldBasedIndexPatternColumn, ReferenceBasedIndexPatternColumn | undefined]
   > = Object.values(state.layers)
     // filter layers without dataView loaded yet
-    .filter(({ indexPatternId }) => dataViews.indexPatterns[indexPatternId])
+    .filter(({ indexPatternId }) => indexPatternId && dataViews.indexPatterns[indexPatternId])
     .flatMap((layer) => {
+      if (!isFormBasedLayer(layer)) {
```

Comment:
can we move this into the `filter` call?

Reply:
doesnt seem to work